### PR TITLE
Log reason for participant close

### PR DESCRIPTION
**Pull request by boks1971**
Please suggest better naming for different scenarios if something comes
to mind.

### CODE REVIEW

File: pkg/rtc/types/interfaces.go [hunk 1]
```diff
@@ -54,6 +51,68 @@ func (m MigrateState) String() string {
...
+	ParticipantCloseReasonDuplicateIdentity
+	ParticipantCloseReasonServiceRequest
+	ParticipantCloseReasonSimulateMigration
+	ParticipantCloseReasonSimulateNodeFailure
+	ParticipantCloseReasonSimulateServerLeave
+)
+
+func (p ParticipantCloseReason) String() string {
+	switch p {
+	case ParticipantCloseReasonSignalRequestLeave:
```
Comment (davidzhao) [inline pkg/rtc/types/interfaces.go:83]: ClientRequestLeave?

File: pkg/rtc/types/interfaces.go [hunk 2]
```diff
@@ -54,6 +51,68 @@ func (m MigrateState) String() string {
...
+	case ParticipantCloseReasonJoinFailed:
+		return "JOIN_FAILED"
+	case ParticipantCloseReasonJoinTimeout:
+		return "JOIN_TIMEOUT"
+	case ParticipantCloseReasonRTCSessionFinish:
+		return "RTC_SESSION_FINISH"
+	case ParticipantCloseReasonDeleteRoom:
+		return "DELETE_ROOM"
+	case ParticipantCloseReasonDisconnected:
+		return "PEER_CONNECTION_DISCONNECTED"
```
Comment (davidzhao) [inline pkg/rtc/types/interfaces.go:94]:
```suggestion
		return "STATE_DISCONNECTED"
```

**Review by davidzhao — APPROVED** [review]

File: pkg/rtc/types/interfaces.go (hunk 1 above)
Comment (boks1971) [reply pkg/rtc/types/interfaces.go:83]: Will do.

File: pkg/rtc/types/interfaces.go (hunk 2 above)
Comment (boks1971) [reply pkg/rtc/types/interfaces.go:94]: Yikes, will fix.

**Review by cnderrauber — APPROVED** [review]
Looks good, thanks!

**Pull request merged**